File: caller.py
from metacall import metacall

def run_py_caller():
    js_result = metacall("js_multiply", 4, 5)
    rb_result = metacall("rb_shout", "world")

    print(f"[py→js] {js_result}")
    print(f"[py→rb] {rb_result}")

    # Returning a dictionary ensures FaaS can send a proper JSON response to the user
    return {"js_call": js_result, "rb_call": rb_result}

File: caller.js
const { metacall } = require("metacall");

function run_js_caller() {
    const py_result = metacall("py_add", 10, 20);
    const rb_result = metacall("rb_shout", "metacall");

    console.log(`[js→py] ${py_result}`);
    console.log(`[js→rb] ${rb_result}`);

    // Returning an object ensures FaaS can send a proper JSON response
    return { py_call: py_result, rb_call: rb_result };
}

module.exports = { run_js_caller };